src/charm.py

```diff
@@ -177,6 +177,9 @@ def _on_config_changed(self, event: ConfigChangedEvent) -> None:
 
             self.on[self.restart.name].acquire_lock.emit()
 
+        if self.model.get_relation(REL_NAME):
```

Comment:
This will fail for N connections.

Could we loop through all relations, and assert at least one of them is 'kafka-client'?

Reply:
You are right. I'm wondering if doing this will return an empty list or an exception:
```
if self.model.relations[REL_NAME]:
    thing
```

Reply:
That I'm not certain of! It returns a dict-like, maybe a `.get(x, None)`